lte/gateway/python/integ_tests/s1aptests/test_attach_detach_attach_dl_tcp_data_pipelined_tbl0.py

```diff
@@ -0,0 +1,112 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+
+import s1ap_types
+import s1ap_wrapper
+from integ_tests.s1aptests.s1ap_utils import MagmadUtil
+
+
+class TestAttachDetachAttachDlTcpDataPipelinedTbl0(unittest.TestCase):
+
+    def setUp(self):
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper()
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def test_attach_detach_attach_dl_tcp_data_pipelined_tbl0(self):
+        """ Attach, detach, re-attach, send DL TCP data, and detach """
+        self._s1ap_wrapper.configUEDevice(1)
+        req = self._s1ap_wrapper.ue_req
+        print(
+            "************************* Running 1st End to End attach for ",
+            "UE id ", req.ue_id,
+        )
+
+        # Enable the enable5g_features flag to pipelined
+        self._s1ap_wrapper.magmad_util.config_enable5g_features(
+                MagmadUtil.enable5g_features_cmds.ENABLE,
+        )
+
+        # Enable the flag to control ovs table=0 using pipelined
+        self._s1ap_wrapper.magmad_util.config_pipelined_managed_tbl0(
+                MagmadUtil.pipelined_managed_tbl0_cmds.ENABLE,
+        )
+        # To reflect above config restart pipelined service is mandatory
+        self._s1ap_wrapper.magmad_util.restart_services(["pipelined"])
```

Comment:
Please add 30 sec sleep after every service restart in this PR change. Sample code is in testcase: s1aptests/test_attach_detach_with_mme_restart.py

Reply:
Thanks for letting us know this after adding this wait time test cases become success.